api.planx.uk/s3/uploadFile.ts

```diff
@@ -33,13 +34,22 @@ export const uploadPrivateFile = async (
   };
 
   await s3.putObject(params).promise();
+  const fileUrl = await buildFileUrl(key, "private")
 
   return {
     file_type: fileType,
-    key,
+    fileUrl,
   };
 };
 
+// Construct an API URL for the uploaded file
+const buildFileUrl = async (key: string, path: "public" | "private") => {
+  const s3 = s3Factory();
+  const s3Url = await new URL(s3.getSignedUrl('getObject', { Key: key }));
+  const fileUrl = `${process.env.API_URL_EXT}/file/${path}${s3Url.pathname}`
```

Comment:
`pathname` has a leading `/` so this is correct despite looking weird!

Reply:
This commend might be worth of becoming a code-comment.